Cleared warnings with JS Hint

surface2air opened the pull request:
JS linting has been applied to the file "trouble.js" using JS Hint. 

The original file has been renamed to "_trouble.js." This file raised warnings in JS Hint before linting and clean-up. 

The linted and clean-up version is now "trouble.js." It's good to go! :+1:

surface2air committed:
Cleared warnings with JS Hint

jamdon commented on app/js/trouble.js:
```diff
@@ -1,123 +1,70 @@
-'use strict';
-
-(function (gbl, $) {
```
You don't want to modify this part - the code is wrapped in an immediately executing function expression (IIFE) to prevent declaring a bunch of variables in the global namespace.

jamdon commented:
The original code has been modified to the point where it's difficult to point out the additional issues introduced while still addressing the original issues with the code. Fixing the original linting issues doesn't require the level of modification this file currently has. I ran your code through the linting tool but there are still a fair amount of linting errors, many of which are new because of the modifications. Do you want me to grade it as is, or would you prefer to wait until I've covered the assignment at the beginning of the next class?

surface2air committed:
Module 01 - JSHint linting exercise

Completed an updated version of the linting exercise for trouble.js in
Module 01 using JSHint

jamdon commented:
Hi Eric. I appears the latest version of your code hasn't been linted yet. Are you certain you synced up your local repository with the github repository?

surface2air commented:
Hi James,

I am missing a step here and there. 

I have done a sync last week and today, and apparently my local git sync is, well out of sync.

I do not see my updated version (for example, today's version contains my name and today's date: May 11, 2015) up here on Github.

So, until I can get this sorted, here's my updated trouble.js file for the JSHint linting exercise:

Eric

/******************************************************************************************************************/

// MODULE 01 - USING MODULE PATTERN - JSHINTING EXERCISE
// Eric Gould - resyncing, May 11, 2015

(function (gbl, $) {

  'use strict';

  gbl.dropdown = function (options) {

```
    var $dropdown = options.dropdown,
        $toggle = options.toggleButton,
        token = +new Date(),
        ns = options.namespace;
        elList = $('.dropdowns'),
        controlsMegaMenu = options.controlsMegaMenu,
        clickAnywhereToClose = options.clickAnywhereToClose,
        transitionEnd = gbl.utilities.whichTransitionEvent();


    $dropdown.addClass('gbl_dropdown').data('status', 'closed');
    $toggle.addClass('gbl_dropdown_trigger');
    $dropdown.attr('aria-expanded', 'false');
    $toggle.attr('aria-controls', $dropdown.attr('id'));
    $dropdown.wrapInner('<div class="measureHeight"></div>');

    function closeMegaMenu() {
        if (gbl.megaMenu && gbl.megaMenu.small && typeof gbl.megaMenu.small.close === 'function') {
            gbl.megaMenu.small.close();
        }
    }

    function setCloseHandler() {
        $(document).on('click.' + ns, function (e) {
            var $clicked = $(e.target);
            if (!$clicked.is($dropdown) && ($clicked.parents().filter($dropdown).length === 0)) {
                close();
            }
        });
    }

    function removeCloseHandler() {
        $(document).off('click.' + ns);
    }

    function processDropdownEls() {
        var list = Array.prototype.slice.call(elList);
    }

    function setDropdownHeight() {
        $dropdown.height($dropdown.find('.measureHeight').height());
    }

    function close() {
        dateStamp = +new Date();
        setDropdownHeight();
        $dropdown.data("status", 'closed');
        $dropdown.removeClass('gbl_dropdown_active');

  setTimeout(setDropdownHeight, 1000);

        $toggle.removeClass('gbl_dropdown_active');
        $toggle.focus();
        var dateStamp;
        $dropdown.attr('aria-expanded', 'false');
        setTimeout(function () {
            $dropdown.removeClass("no_transition");
            $dropdown.css('height', 0);
        }, 50);
        removeCloseHandler();
        $(document).trigger(ns + 'Close');
    }

    function open() {
        $dropdown.removeClass('no_transition');
        $dropdown.data('status', 'open');
        $dropdown.addClass('gbl_dropdown_active');
        $dropdown.focus();
        $toggle.addClass('gbl_dropdown_active');
        $dropdown.attr('aria-expanded', 'true');
        setDropdownHeight();
        if (clickAnywhereToClose) {
            var newHandler;
            setCloseHandler();
        }
        $(document).trigger(ns + 'Open');
    }

    function toggleDropdown(e) {
        e.preventDefault();
        e.stopPropagation();
        if ($dropdown.data('status') ==='closed') {
            open();
        } else {
            close();
        }
        if (controlsMegaMenu) {
            closeMegaMenu();
  }
    }

    $dropdown.on(transitionEnd, function () {
        if ($dropdown.data('status') === "open") {
            $dropdown.addClass('no_transition');
            $dropdown.css('height', 'auto');
        }
    });

    $toggle.on('click', toggleDropdown);

};
```

}(window.gbl || {}, jQuery));

surface2air committed:
Module 01 - Using Module Pattern - JSHint Linting Exercise

Restarted local version of Github for Mac OS and trying to do another
sync of file (trouble.js selected locally in Github for Mac OS).

surface2air committed:
Module 01 - JSHint Linting Exercise

Re-syncing an updated iteration of trouble.js that has been linted with
JSHint - 14 May 2015. Thanks.

jamdon commented on app/js/trouble.js:
```diff
@@ -1,123 +1,108 @@
+
+		var $dropdown = options.dropdown,
+			$toggle = options.toggleButton,
+			ns = options.namespace;
```
this line should end with a comma, not a semicolon - it makes the subsequent line declare a global variable because it begins a new statement rather than continuing a var declaration.

jamdon commented:
:+1:
